src/main/java/seedu/address/model/person/Person.java

```diff
@@ -118,6 +118,29 @@ public Person withUpdatedPublicAddress(PublicAddress newPublicAddress) {
         return new Person(name, phone, email, address, updatedPublicAddresses, tags);
     }
 
+    /**
+     * Returns a new {@code Person} object with the public addresses from the network removed.
+     *
+     * @param network The network to remove the public address
+     * @param label The label of the public address to be removed
+     * @return A new {@code Person} object with the public addresses removed
+     */
+    public Person withoutPublicAddressByNetworkAndLabel(Network network, String label) {
+        PublicAddressesComposition updatedPublicAddresses = publicAddressesComposition.remove(label, network);
+        return new Person(name, phone, email, address, updatedPublicAddresses, tags);
+    }
+
+    /**
+     * Returns a new {@code Person} object with the public addresses from the network removed.
+     *
+     * @param network The network to remove the public address
+     * @return A new {@code Person} object with the public addresses removed
+     */
+    public Person withoutPublicAddressesByNetwork(Network network) {
+        PublicAddressesComposition updatedPublicAddresses = publicAddressesComposition.remove(network);
+        return new Person(name, phone, email, address, updatedPublicAddresses, tags);
+    }
```

Comment:
Although there's no max length per se, these methods are very verbose and could probably be shortened

Can also consider overloading the method since they perform similar operations (one with an extra label)